### PR TITLE
Add support for scoped-registries

### PR DESCRIPTION
**Pull request by FriggaHel**
Add support for multiple registries

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -6,8 +6,15 @@ export type Suite = {
   }
 }
 
+export type ScoredRegistry = {
```
Comment (mhan83) [inline src/types.ts:9]:
```suggestion
export type ScopedRegistry = {
```

File: tests/unit/src/utils.spec.ts [hunk 2]
```diff
@@ -195,6 +195,29 @@ describe('utils', function () {
...
+      cfg.npm ||= {};
+      cfg.npm.scopedRegistries = [{
+        url: 'http://demo.registry.com/npm-test/',
+        scope: '@saucelabs',
+      }];
+      const loadSpyOn = jest.spyOn(npm, 'configure');
+      await prepareNpmEnv(cfg, nodeCtx);
+      expect(loadSpyOn.mock.calls[loadSpyOn.mock.calls.length - 1]).toMatchSnapshot();
+    });
+    it('should configure scoped-registry with authtentication', async function () {
```
Comment (mhan83) [inline tests/unit/src/utils.spec.ts:209]:
```suggestion
    it('should configure scoped-registry with authentication', async function () {
```

File: src/utils.ts [hunk 3]
```diff
@@ -98,12 +98,30 @@ export function getNpmConfig (runnerConfig: NpmConfigContainer) {
...
+    for (const sr of runnerConfig.npm.scopedRegistries) {
+      cfg[`${sr.scope}:registry`] = sr.url;
+      if (sr.authToken) {
+        let fixedUrl = sr.url;
+        if (fixedUrl.startsWith('http://')) {
+          fixedUrl = sr.url.substring(5);
+        } else if (fixedUrl.startsWith('https://')) {
+          fixedUrl = sr.url.substring(6);
+        }
+        cfg[`${fixedUrl}`] = sr.authToken;
```
Comment (mhan83) [inline src/utils.ts:120]: Docs say you need to suffix with `:_authToken`

```suggestion
        cfg[`${fixedUrl}:_authToken`] = sr.authToken;
```

File: tests/unit/src/utils.spec.ts [hunk 4]
```diff
@@ -195,6 +195,44 @@ describe('utils', function () {
...
+    it('should configure scoped-registry', async function () {
+      const cfg = _.cloneDeep(runCfg);
+      cfg.npm ||= {};
+      cfg.npm.registries = [{
+        url: 'http://demo.registry.com/npm-test/',
+        scope: '@saucelabs',
+      }];
+      const loadSpyOn = jest.spyOn(npm, 'configure');
+      await prepareNpmEnv(cfg, nodeCtx);
+      expect(loadSpyOn.mock.calls[loadSpyOn.mock.calls.length - 1]).toMatchSnapshot();
```
Comment (mhan83) [inline tests/unit/src/utils.spec.ts:207]: Generally not a huge fan of using `toMatchSnapshot`. Puts the test into change detector test territory, e.g. adding new npm configs will cause the test to fail.

**Review by mhan83 — APPROVED** [review]

File: src/types.ts [hunk 5]
```diff
@@ -6,8 +6,15 @@ export type Suite = {
...
 
+export type Registry = {
+  scope?: string;
+  url: string;
+  authToken?: string;
+};
+
 export type NpmConfig = {
   registry?: string;
+  registries?: Registry[];
```
Comment (alexplischke) [inline src/types.ts:17]: So are we intending to deprecate `registry` then? Since `registries` supersedes `registry`.

Comment (FriggaHel) [reply src/types.ts:17]: Yes, as `registries` provides the same level of functionality as `registry`.
It also allow to add authentication for non-scoped registries.

Comment (FriggaHel) [reply src/types.ts:20]: I annotated the `registry` flag as deprecated, and to be use `registries` instead.

**Review by alexplischke — APPROVED** [review]

**Pull request merged**